jit-compiler/src/codegen.rs

```diff
@@ -702,7 +732,7 @@ mod test {
         );
         assert_eq!(
             result,
-            "fn c((i): (ibig::IBig)) -> ibig::IBig { ((i).clone() + (ibig::IBig::from(20_u64)).clone()) }\n\
+            "fn c((i): (ibig::IBig)) -> ibig::IBig { Add::add((i).clone(), (ibig::IBig::from(20_u64)).clone()) }\n\
```

Comment:
are the `clone()`s needed here?

Reply:
I don't think so, because this is a full function and both operands can be moved in.

Reply:
This is juts the result. The clone is added locally, it doesn't know that `i` is only used once, for example. If you do `i + i`, you need the clone.

Reply:
The line is a complete function, you do know it is used only once.